src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -224,18 +224,60 @@ public static void Main(string[] args)
         [ConditionalFact(typeof(WindowsOnly))]
         public void NestedIfStatements()
         {
-            int nestingLevel = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
+            (int nestingLevelLower, int nestingLevelHigher) = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
             {
-                (ExecutionArchitecture.x86, ExecutionConfiguration.Debug) => 310,
-                (ExecutionArchitecture.x86, ExecutionConfiguration.Release) => 1650,
-                (ExecutionArchitecture.x64, ExecutionConfiguration.Debug) => 200,
-                (ExecutionArchitecture.x64, ExecutionConfiguration.Release) => 780,
+                (ExecutionArchitecture.x86, ExecutionConfiguration.Debug) => (310, 400),
+                (ExecutionArchitecture.x86, ExecutionConfiguration.Release) => (1650, 2000),
+                (ExecutionArchitecture.x64, ExecutionConfiguration.Debug) => (200, 300),
+                (ExecutionArchitecture.x64, ExecutionConfiguration.Release) => (780, 1000),
                 _ => throw new Exception($"Unexpected configuration {ExecutionConditionUtil.Architecture} {ExecutionConditionUtil.Configuration}")
             };
 
-            RunTest(nestingLevel, runTest);
+            var parseOptionsThrow = TestOptions.Regular.WithThrowOnInsufficientExecutionStack(true);
+            var parseOptionsNoThrow = TestOptions.Regular;
 
-            static void runTest(int nestingLevel)
+            // Lower nesting level, single stack. No errors.
+            RunTest(
+                nestingLevelLower,
+                nestingLevel =>
+                {
+                    var source = generateSource(nestingLevel);
+                    RunInThread(() =>
+                    {
+                        var comp = CreateCompilation(source, parseOptions: parseOptionsThrow);
+                        comp.VerifyDiagnostics();
+                    });
+                });
+
+            // Higher nesting level, single stack. Expression too complex error.
+            RunTest(
+                nestingLevelHigher,
+                nestingLevel =>
+                {
+                    var source = generateSource(nestingLevel);
+                    RunInThread(() =>
+                    {
+                        var comp = CreateCompilation(source, parseOptions: parseOptionsThrow);
+                        comp.VerifyDiagnostics(
+                            // error CS8078: An expression is too long or complex to compile
+                            Diagnostic(ErrorCode.ERR_InsufficientStack, ""));
+                    });
+                });
+
+            // Higher nesting level, single stack. No errors.
```

Comment:
>single stack [](start = 37, length = 12)

this comment is misleading.  Perhaps it should say something like "multiple stacks if needed".